src/devices/cpu/uml.cpp

```diff
@@ -504,13 +508,26 @@ void uml::instruction::simplify()
 				else if (m_param[2].is_immediate() && m_param[3].is_immediate())
 				{
 					if (m_size == 4)
-						convert_to_mov_immediate(u32(u32(m_param[1].immediate()) * u32(m_param[2].immediate())));
+						convert_to_mov_immediate(u32(u32(m_param[2].immediate()) * u32(m_param[3].immediate())));
 					else if (m_size == 8)
-						convert_to_mov_immediate(u64(u64(m_param[1].immediate()) * u64(m_param[2].immediate())));
+						convert_to_mov_immediate(u64(u64(m_param[2].immediate()) * u64(m_param[3].immediate())));
 				}
 			}
 			break;
 
+		// MULULW: convert simple form to MOV if immediate, or if multiplying by 0
+		case OP_MULULW:
+			if (m_param[1].is_immediate_value(0) || m_param[2].is_immediate_value(0))
+				convert_to_mov_immediate(0);
+			else if (m_param[1].is_immediate() && m_param[2].is_immediate())
+			{
+				if (m_size == 4)
+					convert_to_mov_immediate(u32(u32(m_param[1].immediate()) * u32(m_param[2].immediate())));
+				else if (m_size == 8)
+					convert_to_mov_immediate(u64(u64(m_param[1].immediate()) * u64(m_param[2].immediate())));
+			}
+			break;
```

Comment:
Will the code generated by `convert_to_mov_immediate` set the overflow flag appropriately if a condition code update is requested?

Reply:
The simplify function in uml.cpp only works for opcodes that won't update status flags, so this code is fine.

See: https://github.com/mamedev/mame/blob/master/src/devices/cpu/uml.cpp#L370-L374